src/main/java/com/capitalone/dashboard/collector/SonarCollectorTask.java

```diff
@@ -248,69 +248,87 @@ private void deleteUnwantedJobs(List<SonarProject> latestProjects, List<SonarPro
     private void refreshData(List<SonarProject> sonarProjects, SonarClient sonarClient) {
         long start = System.currentTimeMillis();
         int count = 0;
-
+        int updated = 0;
+        int disabled = 0;
         for (SonarProject project : sonarProjects) {
-            CodeQuality codeQuality = sonarClient.currentCodeQuality(project);
-            if (codeQuality != null && isNewQualityData(project, codeQuality)) {
-                project.setLastUpdated(System.currentTimeMillis());
-                sonarProjectRepository.save(project);
-                codeQuality.setCollectorItemId(project.getId());
-                codeQualityRepository.save(codeQuality);
-                count++;
+            try {
+                CodeQuality codeQuality = sonarClient.currentCodeQuality(project);
+                if (codeQuality != null && isNewQualityData(project, codeQuality)) {
+                    project.setLastUpdated(System.currentTimeMillis());
+                    sonarProjectRepository.save(project);
+                    codeQuality.setCollectorItemId(project.getId());
+                    codeQualityRepository.save(codeQuality);
+                    updated++;
+                }
+            } catch (HttpClientErrorException e) {
+                String compare = "{\"errors\":[{\"msg\":\"Component id \'" + project.getProjectId() + "\' not found\"}]}";
+                if ((e.getStatusCode() == HttpStatus.NOT_FOUND) && (e.getResponseBodyAsString().equals(compare))) {
+                    project.setEnabled(false);
```

Comment:
Make it a bit looser, in case the error messages are slight different: 
String body = e.getResponseBodyAsString();
if (e.getStatusCode()==HttpStatus.NOT_FOUND && body!=null && body.contains("errors") && body.contains("not found")) {
... ....
}